build-system/tasks/compile-expr.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs-extra');
+const gulp = require('gulp');
+const jison = require('jison');
+
+/**
+ * Helper function that uses jison to generate a parser for the input file.
+ * @param {string} path
+ * @param {string} jisonFilename
+ * @param {string} imports
+ * @param {string} parserName
+ * @param {string} jsFilename
+ */
+function compileExpr(path, jisonFilename, imports, parserName, jsFilename) {
+  const bnf = fs.readFileSync(path + jisonFilename, 'utf8');
+  const settings = {
+    type: 'lalr',
+    debug: false,
+    moduleType: 'js',
+    'token-stack': true,
```

Comment:
Continuing https://github.com/ampproject/amphtml/pull/18574#discussion_r223506153:

I don't think this is an acceptable change. Passing this causes the lexer to use an actual stack to process, which causes a lot of array mutations. This will very likely have an impact on runtime performance, which is already pretty bad.

Reply:
Noted. Removed.